nntrainer/src/model_loader.cpp

```diff
@@ -107,11 +117,12 @@ int ModelLoader::loadDatasetConfigIni(dictionary *ini, NeuralNetwork &model) {
   status = parse_and_set("Dataset:LabelData", DATA_LABEL, true);
   NN_RETURN_STATUS();
 
-  /// fixme: #299, #389
   int bufsize = iniparser_getint(ini, "DataSet:BufferSize", model.batch_size);
   ml_logd("buf size: %d", bufsize);
   status = model.data_buffer->setBufSize(bufsize);
   NN_RETURN_STATUS();
+  status = model.data_buffer->setBatchSize(model.batch_size);
```

Comment:
This should be done before :122

Reply:
More Idealy, IMO, setBufSize shouldn't rely one batchSize rather initialize checks if bufsize > batchsize though. I think it is slightly out of topic.

Reply:
> This should be done before :122

It is not necessary. bufsize is already set at line 120 and initialized with model.batch_size. This line sets model.data_buffer.batch_size.

Reply:
Updated.